lib/oidc-client.js

```diff
@@ -384,7 +387,9 @@ OidcClient.prototype = {
         });
     },
 
-    processResponseAsync: function (result, requestState) {
+    processResponseAsync: function (result, requestState, user_state) {
+        user_state = user_state || null;
```

Comment:
You're not pulling the state from the function itself, you're pulling it out of the state cookie.